IS0402Test.py

```diff
@@ -507,31 +412,595 @@ def test_20(self):
-            elif r.status_code == 201:
-                return test.FAIL("Registration API returned wrong HTTP code.")
-            elif r.status_code == 200:
-                return test.PASS()
-            else:
-                return test.FAIL("Registration API returned an unexpected response: {} {}"
-                                 .format(r.status_code, r.text))
+            return test.PASS()
         else:
             return test.FAIL("Version > 1 not supported yet.")
 
-    def test_21(self):
-        """Query API implements pagination"""
+    def check_paged_trait(self, test):
+        """Precondition check that the 'paged' trait applies to the API version under test"""
+
+        api = self.apis[QUERY_API_KEY]
+        if self.is04_query_utils.compare_api_version(api["version"], "v1.1") < 0:
+            raise NMOSTestException(test.NA("This test does not apply to v1.0"))
+        if self.is04_query_utils.compare_api_version(api["version"], "v2.0") >= 0:
+            raise NMOSTestException(test.FAIL("Version > 1 not supported yet."))
 
-        test = Test("Query API implements pagination")
+    def post_sample_nodes(self, test, count, description, labeller = None):
+        """Perform a POST request on the Registration API to register a number of sample nodes"""
 
-        if self.apis[QUERY_API_KEY]["version"] == "v1.0":
-            return test.NA("This test does not apply to v1.0")
+        node_data = deepcopy(self.test_data["node"])
+        node_data["description"] = description
+
+        update_timestamps = []
+        ids = []
+
+        for _ in range(count):
+            ids.append(str(uuid.uuid4()))
+
+            node_data["id"] = ids[-1]
+            self.bump_resource_version(node_data)
+
+            if labeller != None:
+                node_data["label"] = labeller(_)
+
+            # For debugging
+            node_data["tags"]["index"] = [str(_)]
+
+            self.post_resource(test, "node", node_data, 201)
+
+            # Perform a Query API request to get the update timestamp of the most recently POSTed node
+            # Wish there was a better way, as this puts the cart before the horse!
+            # Another alternative would be to use local timestamps, provided clocks were synchronised?
+
+            valid, r = self.do_request("GET", self.query_url + "nodes")
+            if not valid:
+                raise NMOSTestException(test.FAIL("Cannot GET the POSTed sample data. Cannot execute test: "
+                                                  "{}".format(r)))
+            elif r.status_code != 200:
+                raise NMOSTestException(test.FAIL("Cannot GET the POSTed sample data. Cannot execute test: "
+                                                  "{} {}".format(r.status_code, r.text)))
+
+            try:
+                if r.json()[0]["id"] != node_data["id"]:
+                    raise NMOSTestException(test.FAIL("Query API response did not have the most recently POSTed node first"))
```

Comment:
Is it worth checking here if any paging headers are returned first? You won't get a 501 before this point, but if a registry doesn't implement pagination there isn't necessarily an ordering guarantee.

Reply:
> "if a registry doesn't implement pagination there isn't necessarily an ordering guarantee"

😲 Oh my, that hadn't even occurred to me!

Reply:
Resolved by 3837909c27935738d5afe3db8d9be1a73f485394